mrp_bom_version/models/mrp_bom.py

```diff
@@ -0,0 +1,165 @@
+# (c) 2015 Oihane Crucelaegui - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, fields, models
+from odoo.tools import config
+
+
+class MrpBom(models.Model):
+    _inherit = "mrp.bom"
+
+    def _compute_old_versions(self):
+        for bom in self:
+            parent = bom.parent_bom
+            old_version = self.env["mrp.bom"]
+            while parent:
+                old_version |= parent
+                parent = parent.parent_bom
+            bom.old_versions = [(6, 0, old_version.ids)]
+
+    def _default_active(self):
+        """Needed for preserving normal flow when testing other modules."""
+        res = False
+        if config["test_enable"]:
+            res = not bool(self.env.context.get("test_mrp_bom_version"))
+        return res
+
+    def _default_state(self):
+        """Needed for preserving normal flow when testing other modules."""
+        res = "draft"
+        if config["test_enable"] and not self.env.context.get("test_mrp_bom_version"):
+            res = "active"
+        return res
+
+    active = fields.Boolean(
+        default=_default_active, readonly=True, states={"draft": [("readonly", False)]}
+    )
+    historical_date = fields.Date(readonly=True, copy=False)
+    state = fields.Selection(
+        selection=[
+            ("draft", "Draft"),
+            ("active", "Active"),
+            ("historical", "Historical"),
+        ],
+        string="Status",
+        index=True,
+        readonly=True,
+        default=_default_state,
+        copy=False,
+    )
+    product_tmpl_id = fields.Many2one(
+        readonly=True, states={"draft": [("readonly", False)]}
+    )
+    product_id = fields.Many2one(readonly=True, states={"draft": [("readonly", False)]})
+    product_qty = fields.Float(readonly=True, states={"draft": [("readonly", False)]})
+    code = fields.Char(states={"historical": [("readonly", True)]})
+    type = fields.Selection(states={"historical": [("readonly", True)]})
+    company_id = fields.Many2one(states={"historical": [("readonly", True)]})
+    product_uom_id = fields.Many2one(states={"historical": [("readonly", True)]})
+    bom_line_ids = fields.One2many(
+        readonly=True, states={"draft": [("readonly", False)]}
+    )
+    byproduct_ids = fields.One2many(
+        readonly=True, states={"draft": [("readonly", False)]}
+    )
+    sequence = fields.Integer(states={"historical": [("readonly", True)]})
+    operation_ids = fields.One2many(
+        readonly=True, states={"draft": [("readonly", False)]}
+    )
+    ready_to_produce = fields.Selection(states={"historical": [("readonly", True)]})
+    picking_type_id = fields.Many2one(states={"historical": [("readonly", True)]})
+    consumption = fields.Selection(states={"historical": [("readonly", True)]})
+    version = fields.Integer(
+        states={"historical": [("readonly", True)]}, copy=False, default=1
+    )
+    parent_bom = fields.Many2one(
```

Comment:
Suggestion: rename field `parent_bom` to `previous_bom`.  The name `parent_bom` seems confusing, given the hierarchical nature of a bill of materials.

Reply:
Thanks for the suggestion, change done

Reply:
@matt454357 @kaycross : is it ok to have an aprove? I would like to merge it but I would need more aprovals.
Thank you!!!